lib/TutorialServo/TutorialServo.cpp

```diff
@@ -0,0 +1,41 @@
+#include "TutorialServo.h"
+
+
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees = 180.0, float minPulsewidthInMs = 1,
+                             float maxPulsewidthInMs = 2) 
+	: _servoPwmOut(servoPin),
+      _minPulseWidthInMs(minPulsewidthInMs),
+      _maxPulseWidthInMs(maxPulsewidthInMs),
+      _servoRangeInDegrees(servoRangeInDegrees) 
+{
+	_minPulseWidthInMs = _minPulseWidthInMs / 1000;
+	_maxPulseWidthInMs = _maxPulseWidthInMs / 1000;
+	//default period of servo is 50Hz or 20 milliseconds 
+	_servoPwmOut.period(0.020);
+}
+
+void TutorialServo::setAngleRangeInDegrees(float degrees) {
+	_servoRangeInDegrees = degrees; 
+}
+
+void TutorialServo::setPulseWidthRangeInMs(float minPulseWidthInMs, float maxPulseWidthInMs) {
+	_minPulseWidthInMs = minPulseWidthInMs/1000;
+	_maxPulseWidthInMs = maxPulseWidthInMs/1000;
+}
+void TutorialServo::setServoPositionInDegrees(float degrees) {
+	
+	float value;

```

Comment:
one improvement to ur code can make ur code more concise. u have 2 functions (setdegrees and setpercent). instead of having them be defined seperately and have similar code inside both of them, it's much nicer to have once call the other. for example, setdegrees can first divide the input with the range and just call setpercent.